reana/reana_dev/utils.py

```diff
@@ -790,3 +792,24 @@ def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
 
     return wrapper
+
+
+def validate_directory(ctx, param, target_directory):
+    """Validate if directory is valid for first time cloning."""
+    if target_directory:
+        target_directory = os.path.realpath(target_directory)
+
+        if not os.path.isdir(target_directory):
+            message = "[ERROR] Directory {0} does not exist. Exiting.".format(
+                target_directory
+            )
+            click.echo(click.style(message, fg="red"), err=True)
+            ctx.exit(1)
+
+        if len(os.listdir(target_directory)) != 0:
+            message = "[ERROR] Directory {0} is not empty. Cloning aborted.".format(
+                target_directory
+            )
+            click.echo(click.style(message, fg="red"), err=True)
+            ctx.exit(1)
```

Comment:
Does the target directory have to be empty?
Imagine that one wants to clone the repo for the first time in `~/my_reana`. First, they would need to clone at least the `reana` repo to `pip install` it and be able to use the `reana-dev` tool. But then, when running `reana-dev git-clone` they would get this error message if they want to clone it in the same directory. Does it make sense?

Reply:
They won't actually clone "reana" repository first, since people are instructed to do first:

```console
          $ # install reana-dev developer helper script
          $ pip install git+git://github.com/reanahub/reana.git#egg=reana
```

see `reana-dev --help`.

So, they get the helper script in this way, and then they clone the whole shebang into an empty repository.

(And when they run the script next time, the original helper script will be overwritten by newly cloned "reana" one.)

Reply:
Mmh I think the CLI is expecting to find the `reana` folder somehow. I'm currently following those docs and I get the following traceback:

```console
$ reana-dev git-clone -c ALL -u mvidalgarcia

fatal: not a git repository (or any of the parent directories): .git
Traceback (most recent call last):
  File "/Users/marco/.virtualenvs/r/bin/reana-dev", line 8, in <module>
    sys.exit(reana_dev())
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/reana/reana_dev/git.py", line 375, in git_clone
    os.chdir(get_srcdir())
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/reana/reana_dev/utils.py", line 144, in get_srcdir
    reana_srcdir = find_reana_srcdir()
  File "/Users/marco/.virtualenvs/r/lib/python3.9/site-packages/reana/reana_dev/utils.py", line 122, in find_reana_srcdir
    subprocess.check_output("git rev-parse --show-toplevel", shell=True)
  File "/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'git rev-parse --show-toplevel' returned non-zero exit status 128.
```

Note the `find_reana_srcdir` call.

Reply:
No, in fact that is the whole initial problem, it looks for reana resources to clone reana resources. It happens because as @tiborsimko mentioned in [this comment](https://github.com/reanahub/reana/pull/523#discussion_r666749724), when there is no `--target-directory` it falls again in the same old behavior (and issue). Maybe we can make `--target-directory` a required value...

Reply:
> it looks for reana resources to clone reana resources.

This looks wrong to me...

If we don't consider `--target-directory` at all, what is the purpose of calling `find_reana_scrdir` if the command is supposed to clone `reana` repo as well?

Reply:
I agree with you, what do you @mvidalgarcia and @tiborsimko think of the workaround in my last commit?. In that case we won't call `find_reana_scrdir` and simply won't clone `reana` if it is already there